recipes/mold/all/conanfile.py

```diff
@@ -65,6 +65,8 @@ def validate(self):
             raise ConanInvalidConfiguration("Clang version 12 or higher required")
         if self.settings.compiler == "apple-clang" and "armv8" == self.settings.arch :
             raise ConanInvalidConfiguration(f'{self.name} is still not supported by Mac M1.')
+        if Version(self.version) >= "2.33.0" and self.settings.compiler == "apple-clang" and Version(self.settings.compiler.version) < "14":
```

Comment:
Let's keep this check to this specific version only, so that when a new version is added, we can check if the new version fixed the issue :)

```suggestion
        if Version(self.version) == "2.33.0" and self.settings.compiler == "apple-clang" and Version(self.settings.compiler.version) < "14":
```